src/oracles/CoinbaseOracle.sol

```diff
@@ -8,48 +8,45 @@ import {IAggregatorV3SourceCoinbase} from "../interfaces/coinbase/IAggregatorV3S
  * @notice A smart contract that serves as an oracle for price data reported by a designated reporter.
  */
 contract CoinbaseOracle is IAggregatorV3SourceCoinbase {
-    address immutable reporter;
+    uint8 public immutable decimals = 6;
 
-    uint8 public immutable decimals;
+    struct RoundData {
+        int256 answer;
+        uint256 timestamp;
+    }
 
     struct PriceData {
         uint80 lastRoundId;
-        mapping(uint80 => int256) roundAnswers;
-        mapping(uint80 => uint256) roundTimestamps;
+        mapping(uint80 => RoundData) rounds;
     }
 
-    mapping(string => PriceData) private prices;
+    mapping(string => PriceData) internal prices;
 
     event PricePushed(string indexed ticker, uint80 indexed roundId, int256 price, uint256 timestamp);
 
     /**
-     * @notice Constructor to initialize the CoinbaseOracle contract.
-     * @param _decimals The number of decimals in the reported price.
-     * @param _reporter The address of the reporter allowed to push price data.
+     * @notice Returns the address of the reporter.
+     * @return The address of the reporter.
      */
-    constructor(uint8 _decimals, address _reporter) {
-        decimals = _decimals;
-        reporter = _reporter;
+    function reporter() public view virtual returns (address) {
+        return 0xfCEAdAFab14d46e20144F48824d0C09B1a03F2BC;
```

Comment:
better leave as immutable. If mock needs override then can do that in mock constructor